perf-event/src/lib.rs

```diff
@@ -432,6 +451,137 @@ pub struct CountAndTime {
     pub time_running: u64,
 }
 
+bitflags::bitflags! {
+    /// Specifies which fields to include in the sample.
+    ///
+    /// These fields will be recorded in the [`Sampler`] output buffer.
+    ///
+    /// [`Sampler`]: crate::Sampler
+    #[derive(Default)]
+    pub struct Sample : u64 {
+        /// Record the instruction pointer.
+        const IP = sys::bindings::PERF_SAMPLE_IP;
+
+        /// Record the process and thread IDs.
+        const TID = sys::bindings::PERF_SAMPLE_TID;
+
+        /// Record a timestamp.
+        const TIME = sys::bindings::PERF_SAMPLE_TIME;
+
+        /// Record an address, if applicable.
+        const ADDR = sys::bindings::PERF_SAMPLE_ADDR;
+
+        /// Record counter values for all events in a group, not just the
+        /// group leader.
+        const READ = sys::bindings::PERF_SAMPLE_READ;
+
+        /// Record a unique ID for the opened event's group leader.
+        const CALLCHAIN = sys::bindings::PERF_SAMPLE_CALLCHAIN;
+
+        /// Record CPU number.
+        const ID = sys::bindings::PERF_SAMPLE_ID;
+
+        /// Record the callchain (stack backtrace).
+        const CPU = sys::bindings::PERF_SAMPLE_CPU;
+
+        /// Record the current sampling period.
+        const PERIOD = sys::bindings::PERF_SAMPLE_PERIOD;
+
+        /// Record a unique ID for the opened event. Unlike [`ID`] an actual ID
+        /// is returned, not the group leader.
+        ///
+        /// This ID is the same as the one returned by [`Counter::id`]
+        ///
+        /// [`ID`]: Self::ID
+        /// [`Counter::id`]: crate::Counter::id
+        const STREAM_ID = sys::bindings::PERF_SAMPLE_STREAM_ID;
+
+        /// Record additional data, if applicable. Usually returned by
+        /// tracepoint events.
+        const RAW = sys::bindings::PERF_SAMPLE_RAW;
+
+        /// Provides a record of recent branches, as provided by CPU branch
+        /// sampling hardware (such as Intel Last Branch Record). Not all
+        /// hardware supports this feature.
+        ///
+        /// Available since Linux 3.4.
```

Comment:
I love having all this documentation - but we can't just copy the text from `perf_event_open`, because that is [licensed under the GPL v2](https://git.kernel.org/pub/scm/docs/man-pages/man-pages.git/tree/man2/perf_event_open.2?id=4c1c52748a06141996fccbf23fb8d9e15cba11e8#n3), which is incompatible with this crate's licensing.

I see that I made the same mistake in the documentation in `events.rs`. I'll have to take that out, too.

Trying to rewrite such detailed documentation in our own words will take a huge amount of time, and will likely introduce errors. I think our best option is to simply use names in our API that are easy to look up in the man page, and repeatedly link to `http://man7.org/linux/man-pages/man2/perf_event_open.2.html`.